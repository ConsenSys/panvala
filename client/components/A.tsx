import styled from 'styled-components';
import { color, typography } from 'styled-system';

const A: any = styled.a`
  color: inherit;
  cursor: pointer;
  ${color};
  ${({ bold }) => bold && 'font-weight: bold'};
`;

export default A;
