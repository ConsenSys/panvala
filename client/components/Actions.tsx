import * as React from 'react';
import styled from 'styled-components';
import Button from './Button';
import { COLORS } from '../styles';

const ActionsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 2rem;
  font-family: 'Roboto';
`;

const FlexWrapper = styled.div`
  display: flex;
`;
const ActionHelpMessage = styled.div`
  font-size: 0.75rem;
  width: 16rem;
  text-align: right;
  margin-right: 1rem;
  color: ${COLORS.grey3};
`;

const Actions = ({ handleClick, actionText, handleBack }): any => (
  <ActionsWrapper>
    <FlexWrapper>
      <Button large onClick={handleBack}>
        Back
      </Button>
      <Button large type="default" onClick={handleClick}>
        {actionText}
      </Button>
    </FlexWrapper>
    <ActionHelpMessage>
      This will redirect to a separate MetaMask window to confirm your transaction.
    </ActionHelpMessage>
  </ActionsWrapper>
);

export default Actions;
