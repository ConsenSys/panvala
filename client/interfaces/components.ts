import * as React from 'react';

export interface StatelessPage<P = {}> extends React.SFC<P> {
  getInitialProps?: (ctx: any) => Promise<P>;
}

export interface IParameterChangesObject {
  [key: string]: {
    oldValue: string;
    newValue: string;
  };
}

export interface IGovernanceSlateFormValues {
  email: string;
  title: string;
  firstName: string;
  lastName?: string;
  organization?: string;
  summary: string;
  recommendation: string;
  parameters: IParameterChangesObject;
  stake: string;
}
