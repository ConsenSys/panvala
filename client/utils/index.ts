// Re-export Panvala utils to be used throughout
export { default as panvala_utils } from '../../packages/panvala-utils';
