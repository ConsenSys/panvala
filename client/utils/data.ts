import { IProposal, ISlate } from '../interfaces';
import { statuses } from './status';

export const proposalsArray: IProposal[] = [
  {
    id: 2,
    firstName: 'Alice',
    lastName: 'Doe',
    email: 'alice@email.com',
    title: 'K Semantics for Web Assembly',
    summary:
      'A proposal to make Ethereum safer by enabling developers to formally verify smart contracts that run on Web Assembly.',
    tokensRequested: '200000',
    category: 'GRANT',
    totalBudget: '100 for this. 200 for that.',
    otherFunding: 'nope!',
    projectPlan: 'plan',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
    awardAddress: '0xd09cc3bc67e4294c4a446d8e4a2934a921410ed7',
  },
  {
    id: 3,
    firstName: 'Bob',
    lastName: 'Doe',
    email: 'bob@email.com',
    title: 'Registry Builder Vote Weighting',
    summary:
      'A proposal to make Ethereum safer by improving a reusable library for token-curated registries.',
    tokensRequested: '175000',
    category: 'GRANT',
    totalBudget: '100 for this. 200 for that.',
    otherFunding: 'nope!',
    projectPlan: 'plan',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
    awardAddress: '0xd09cc3bc67e4294c4a446d8e4a2934a921410ed7',
  },
  {
    id: 4,
    firstName: 'Cat',
    lastName: 'Doe',
    email: 'cat@email.com',
    title: 'Tidbit Dynamic Oracles',
    summary: 'A proposal to make Ethereum safer by improving a reusable library for oracles.',
    tokensRequested: '150000',
    category: 'GRANT',
    totalBudget: '100 for this. 200 for that.',
    otherFunding: 'nope!',
    projectPlan: 'plan',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
    awardAddress: '0xd09cc3bc67e4294c4a446d8e4a2934a921410ed7',
  },
  {
    id: 5,
    firstName: 'Dog',
    lastName: 'Doe',
    email: 'dog@email.com',
    title: 'Infrastructure Monitoring',
    summary:
      'A proposal to make Ethereum safer by monitoring high value systems and assets on Ethereum.',
    tokensRequested: '200000',
    category: 'GRANT',
    totalBudget: '100 for this. 200 for that.',
    otherFunding: 'nope!',
    projectPlan: 'plan',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
    awardAddress: '0xd09cc3bc67e4294c4a446d8e4a2934a921410ed7',
  },
];

export const slatesArray: ISlate[] = [
  {
    id: 0,
    // metadataHash: 'QmRZxt2b1FVZPNqd8hsiykDL3TdBDeTSPX9Kv46HmX4Gx1',
    category: 'GRANT',
    status: statuses.PENDING_VOTE,
    deadline: 1539044131,
    title: 'Panvala Award Committee',
    description:
      'In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla ...',
    owner: 'Guy Reid',
    ownerAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
    organization: 'Panvala',
    incumbent: true,
    proposals: proposalsArray,
    requiredStake: '500',
  },
  {
    id: 1,
    // metadataHash: 'QmRZxt2b1FVZPNqd8hsiykDL3TdBDeTSPX9Kv46HmX4Gx2',
    category: 'GRANT',
    status: statuses.PENDING_TOKENS,
    deadline: 1539044131,
    title: 'Grant Awards',
    description:
      'In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla sodales ultrices tellus sodales consectetur.Nullam id mattis velit.',
    owner: 'Marcus Berten',
    ownerAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
    organization: 'Grant Awards Group',
    incumbent: false,
    proposals: proposalsArray,
    requiredStake: '500',
  },
  // {
  //   id: 'QmRZxt2b1FVZPNqd8hsiykDL3TdBDeTSPX9Kv46HmX4Gx3',
  //   category: 'GOVERNANCE',
  //   status: statuses.PENDING_TOKENS,
  //   deadline: false,
  //   title: 'Panvala Dev',
  //   description:
  //     'In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla sodales ultrices tellus sodales consectetur.Nullam id mattis velit.',
  //   owner: 'Amber Gilbert',
  //   ownerAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
  //   organization: 'Panvala',
  //   incumbent: false,
  //   proposals: proposalsArray,
  //   requiredStake: '500',
  // },
  // {
  //   id: 'QmRZxt2b1FVZPNqd8hsiykDL3TdBDeTSPX9Kv46HmX4Gx4',
  //   category: 'GOVERNANCE',
  //   status: statuses.PENDING_VOTE,
  //   deadline: 1539044231,
  //   title: 'Governance Change',
  //   description:
  //     'In tempus sem orci, eu auctor mi finibus eu. Nam mi risus, pretium ut laoreet fermentum, cursus nec mi. Nulla sodales ultrices tellus sodales consectetur.Nullam id mattis velit.',
  //   owner: 'Victoria Johnson',
  //   ownerAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
  //   organization: 'Governance Group',
  //   incumbent: false,
  //   proposals: proposalsArray,
  //   requiredStake: '500',
  // },
];
