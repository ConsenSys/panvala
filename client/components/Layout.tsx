import * as React from 'react';
import Head from 'next/head';
import styled from 'styled-components';
import { ToastContainer, toast } from 'react-toastify';
import { format } from 'date-fns';
import { AxiosResponse } from 'axios';

import Header from './Header';
import { IProposal, ISlate, IAppContext } from '../interfaces';
import { getAllProposals, getAllSlates } from '../utils/api';
import { baseToConvertedUnits } from '../utils/format';

export const AppContext: React.Context<IAppContext> = React.createContext<IAppContext>({
  slates: [],
  proposals: [],
  currentBallot: {
    startDate: 0,
    votingOpenDate: 0,
    votingCloseDate: 0,
    finalityDate: 0,
  },
  notifications: [],
});

export const LayoutWrapper = styled.div`
  font-family: 'Roboto';
  min-height: 100vh;
  margin: 2em 10em;
`;

type IProps = {
  title: string;
  children: any;
};

export default class Layout extends React.Component<IProps, IAppContext> {
  readonly state: IAppContext = {
    slates: [],
    proposals: [],
    currentBallot: {
      startDate: 0,
      votingOpenDate: 0,
      votingCloseDate: 0,
      finalityDate: 0,
    },
    notifications: [{ action: 'Welcome to Panvala!' }],
    onNotify: () => this.handleNotify,
    onRefreshProposals: () => this.handleRefreshProposals(),
    onRefreshSlates: () => this.handleRefreshSlates(),
    onHandleNotification: (n: any) => this.handleNotification(n),
  };

  constructor(props: IProps) {
    super(props);
    this.handleNotify = this.handleNotify.bind(this);
    this.handleRefreshProposals = this.handleRefreshProposals.bind(this);
    this.handleRefreshSlates = this.handleRefreshSlates.bind(this);
    this.handleNotification = this.handleNotification.bind(this);
  }

  // runs once, onload
  async componentDidMount() {
    // const slatesFromIpfs: any[] = await Promise.all(slateMultihashes.map(mh => ipfsGetData(mh)));
    // const slatesWithMHs = slates.map((s, i) => ({ ...s, hash: slateMultihashes[i] }));
    let slates: ISlate[] = await this.handleGetAllSlates();
    const proposals: IProposal[] | AxiosResponse = await this.handleGetAllProposals();

    let proposalData: IProposal[] = [];
    // sort proposals by createdAt
    if (Array.isArray(proposals)) {
      const sortedProposals = proposals.sort((a: IProposal, b: IProposal) => {
        const timestampA = format(a.createdAt, 'x');
        const timestampB = format(b.createdAt, 'x');
        return parseInt(timestampA) - parseInt(timestampB);
      });
      proposalData = sortedProposals;
    }

    const oneWeekSeconds = 604800;
    // Epoch 3
    // beginning of week 1 (2/1)
    const epochStartDate = 1549040401; // gateKeeper.functions.currentBatchStart()
    // end of week 11 (4/19)
    const week11EndDate = epochStartDate + oneWeekSeconds * 11; // 1555689601
    // end of week 12 (4/26)
    const week12EndDate = week11EndDate + oneWeekSeconds;
    // end of week 13 (5/3)
    const week13EndDate = week12EndDate + oneWeekSeconds;

    // prettier-ignore
    this.setState({
      slates,
      proposals: proposalData,
      currentBallot: {
        startDate: epochStartDate,
        votingOpenDate: week11EndDate,
        votingCloseDate: week12EndDate,
        finalityDate: week13EndDate,
      },
    });
  }

  handleNotify(note: string, custom: string) {
    if (custom) {
      (toast as any)[custom](note);
    } else {
      toast(note);
    }
  }

  async handleGetAllProposals() {
    let proposals: IProposal[] | AxiosResponse = await getAllProposals();

    // convert tokensRequested: base -> human
    if (Array.isArray(proposals)) {
      proposals = proposals.map((p: any) => {
        p.tokensRequested = baseToConvertedUnits(p.tokensRequested, 18);
        return p;
      });
    } else {
      throw new Error('invalid proposals type -- AxiosResponse');
    }

    return proposals;
  }

  async handleRefreshProposals() {
    const proposals: IProposal[] = await this.handleGetAllProposals();
    return this.setState({ proposals });
  }

  async handleGetAllSlates() {
    const slates: ISlate[] | AxiosResponse = await getAllSlates();
    console.log('slates:', slates);
    if (Array.isArray(slates)) {
      return slates;
    }

    return [];
  }

  async handleRefreshSlates() {
    const slates: ISlate[] = await this.handleGetAllSlates();
    return this.setState({ slates });
  }

  handleNotification(notification: any) {
    this.setState({
      notifications: [...this.state.notifications, notification],
    });
  }

  render() {
    const { children, title }: IProps = this.props;
    console.log('Layout state:', this.state);

    return (
      <LayoutWrapper>
        <Head>
          <title>{title}</title>
        </Head>

        <Header notifications={this.state.notifications} />
        <AppContext.Provider value={this.state}>{children}</AppContext.Provider>
        <ToastContainer
          position="bottom-right"
          autoClose={8000}
          hideProgressBar={true}
          newestOnTop={false}
          rtl={false}
          draggable={false}
          closeOnClick
          pauseOnHover
        />
      </LayoutWrapper>
    );
  }
}
