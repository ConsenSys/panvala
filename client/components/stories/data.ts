import Router from 'next/router';
import range from 'lodash/range';
import uuid from 'uuid/v4';
import { utils } from 'ethers';
import { ballotDates } from '../../utils/voting';
import { ISlate, IProposal } from '../../interfaces';
import '../../globalStyles.css';

// https://github.com/zeit/next.js/issues/1827#issuecomment-323314141
(Router as any).router = {
  push: () => {},
};

export const epochStartDate = 1541178000; // Nov 2, 2018 17:00:00 GMT
export const currentBallot = ballotDates(epochStartDate);
export const baseBallot = {
  epochNumber: 0,
  startDate: 1541178000,
  initialSlateSubmissionDeadline: 1544504400,
  votingOpenDate: 1547830800,
  votingCloseDate: 1548435600,
  finalityDate: 1549040400,
  slateSubmissionDeadline: { GRANT: 0, GOVERNANCE: 0 },
};

export const proposals: IProposal[] = [
  {
    id: 0,
    category: 'GRANT',
    firstName: 'Amanda',
    lastName: 'Crypto',
    title: 'A great project',
    tokensRequested: '5000000000',
    summary:
      "Plasma is a family of protocols which allow individuals to easily deploy high-throughput, secure blockchains. A smart contract on Ethereum’s main chain can ensure that users’ funds are secure, even if the “plasma chain” acts fully maliciously. This eliminates the need for a trusted pegging mechanism like that of sidechains. Plasma chains are non-custodial, allowing the prioritization of scalability without sacrificing security. We’ve devised a new architecture for building Plasma apps on one generalized plasma chain. It establishes a clean separation between the plasma layer and the application layer. We will publish a generalized Plasma predicate contract framework, which allows for upgradeability and composability of plasma contracts. Since plasma research moves so quickly, we realized we needed to develop an architecture that allowed for maximal modularity, to prevent vast chunks of code from being thrown away with each new research discovery. With that framework, we will refactor our existing codebase for secure payments using predicates. We want to use the generalized Plasma research we did and put it to the test in our codebase refactor following the launch of our testnet on January 31. Dogfooding is a critical part of the work we do, as there's no point to open sourcing the codebase if it's not readable or easy to use.",
    awardAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
  },
];

export const unstakedSlate: ISlate = {
  id: 0,
  category: 'GRANT',
  status: 0,
  owner: 'John Doe',
  recommender: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
  organization: 'Team Recommender',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer eu nibh molestie, auctor ligula a, faucibus ante. Morbi dapibus enim in vulputate congue. Mauris feugiat gravida nibh, sed pellentesque eros pellentesque eu. Sed rutrum vitae magna sed aliquet. Suspendisse facilisis vulputate lobortis. Vestibulum sed dolor eu mi molestie pharetra. Duis ut diam aliquam, molestie erat non, scelerisque ligula. Curabitur accumsan ipsum pellentesque posuere ornare. Sed vulputate cursus accumsan. Morbi efficitur dictum magna, a imperdiet mauris aliquet vitae.',
  proposals,
  requiredStake: utils.bigNumberify('300000000000000000000'),
  verifiedRecommender: false,
  incumbent: false,
  staker: '0x0000000000000000000000000000000000000000',
  epochNumber: 1,
};

export function makeBallot(options?: any, prototype?: any) {
  return {
    ...baseBallot,
    ...prototype,
    ...options,
  };
}

export function makeSlate(options: any, prototype?: ISlate) {
  return {
    ...unstakedSlate,
    ...prototype,
    ...options,
    id: uuid(),
  };
}

export function makeSlates(num: number, prototype: ISlate) {
  return range(1, num).map(() => makeSlate({}, prototype));
}

export const acceptedGrantSlate = makeSlate({
  verifiedRecommender: true,
  status: 3,
  epochNumber: 0,
});

export const rejectedGrantSlate = makeSlate(
  {
    status: 2,
  },
  acceptedGrantSlate
);

export const acceptedGovSlate = makeSlate(
  {
    category: 'GOVERNANCE',
    verifiedRecommender: true,
    status: 3,
  },
  acceptedGrantSlate
);

export const rejectedGovSlate = makeSlate(
  {
    status: 2,
  },
  acceptedGovSlate
);
