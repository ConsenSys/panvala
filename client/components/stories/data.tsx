import Router from 'next/router';
import { utils } from 'ethers';
import { ballotDates } from '../../utils/status';
import { ISlate, IProposal } from '../../interfaces';
import '../../globalStyles.css';

// https://github.com/zeit/next.js/issues/1827#issuecomment-323314141
(Router as any).router = {
  push: () => {},
  prefetch: () => {},
};

export const epochStartDate = 1549040401;
export const currentBallot = ballotDates(15499990);

export const proposals: IProposal[] = [
  {
    id: 0,
    firstName: 'Amanda',
    lastName: 'Crypto',
    title: 'A great project',
    tokensRequested: '5000000000',
    summary: 'All sorts of amazing things',
    awardAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
    projectTimeline: 'timeline',
    teamBackgrounds: 'backgrounds',
  },
];

export const unstakedSlate: ISlate = {
  id: 0,
  category: 'GRANT',
  status: 0,
  deadline: currentBallot.votingOpenDate,
  title: 'Some slate',
  owner: 'John Doe',
  recommenderAddress: '0xd115bffabbdd893a6f7cea402e7338643ced44a6',
  organization: 'Team Recommender',
  description:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Integer eu nibh molestie, auctor ligula a, faucibus ante. Morbi dapibus enim in vulputate congue. Mauris feugiat gravida nibh, sed pellentesque eros pellentesque eu. Sed rutrum vitae magna sed aliquet. Suspendisse facilisis vulputate lobortis. Vestibulum sed dolor eu mi molestie pharetra. Duis ut diam aliquam, molestie erat non, scelerisque ligula. Curabitur accumsan ipsum pellentesque posuere ornare. Sed vulputate cursus accumsan. Morbi efficitur dictum magna, a imperdiet mauris aliquet vitae.',
  proposals,
  requiredStake: utils.bigNumberify('300000000000000000000'),
  verifiedRecommender: false,
};
