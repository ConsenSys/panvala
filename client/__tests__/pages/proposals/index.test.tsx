import * as React from 'react';
import { render } from 'react-testing-library';
import 'jest-dom/extend-expect';

import Proposals from '../../../pages/proposals';
import { proposalsArray } from '../../../utils/data';
import { AppContext } from '../../../components/Layout';

describe('Slates', () => {
  it('should render correctly Slates component', () => {
    const proposals = render(
      <AppContext.Provider value={{ proposals: proposalsArray }}>
        <Proposals />
      </AppContext.Provider>
    );

    expect(proposals).toMatchSnapshot();
  });
});
