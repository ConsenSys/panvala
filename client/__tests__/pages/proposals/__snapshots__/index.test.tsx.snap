// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slates should render correctly Slates component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="flex justify-between"
        >
          <div
            class="flex"
          >
            <h1
              class="mr3 RouteTitle-ehvjdd-0 fHYele"
            >
              Proposals
            </h1>
            <a
              class="link flex items-center"
              href="/proposals/create"
            >
              <div
                class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
              >
                <button
                  class="Button__StyledButton-sc-1vedn6s-0 izJlFT"
                  type="default"
                >
                  Add Proposal
                </button>
              </div>
            </a>
          </div>
          <div
            class="Deadline__Wrapper-sc-1a3aqlz-1 gtXKcM"
          >
            <div
              class="Tag__Wrapper-l3n6sk-1 ceKKQx"
            >
              <div
                class="Deadline__StyledDeadline-sc-1a3aqlz-0 jpSHwD Tag__StyledTag-l3n6sk-0 cZzgaW"
              >
                PROPOSAL DEADLINE Invalid date EST
              </div>
            </div>
          </div>
        </div>
        <div
          class="proposals__CardsWrapper-sc-125pbjx-0 cOJbkJ"
        >
          <div
            class="Card__Wrapper-h3ymxi-0 gjicgu"
          >
            <div
              class="flex"
            >
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Tag__StyledTag-l3n6sk-0 cZzgaW"
                >
                  GRANT PROPOSAL
                </div>
              </div>
            </div>
            <div
              class="Card__CardTitle-h3ymxi-1 bSTxPd"
            >
              K Semantics for Web Assembly
            </div>
            <div
              class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
            >
              200000 Tokens Requested
            </div>
            <div
              class="Card__CardDescription-h3ymxi-3 iIGibR"
            >
              A proposal to make Ethereum safer by enabling developers to formally verify smart contracts that run on Web Assembly.
            </div>
          </div>
          <div
            class="Card__Wrapper-h3ymxi-0 gjicgu"
          >
            <div
              class="flex"
            >
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Tag__StyledTag-l3n6sk-0 cZzgaW"
                >
                  GRANT PROPOSAL
                </div>
              </div>
            </div>
            <div
              class="Card__CardTitle-h3ymxi-1 bSTxPd"
            >
              Registry Builder Vote Weighting
            </div>
            <div
              class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
            >
              175000 Tokens Requested
            </div>
            <div
              class="Card__CardDescription-h3ymxi-3 iIGibR"
            >
              A proposal to make Ethereum safer by improving a reusable library for token-curated registries.
            </div>
          </div>
          <div
            class="Card__Wrapper-h3ymxi-0 gjicgu"
          >
            <div
              class="flex"
            >
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Tag__StyledTag-l3n6sk-0 cZzgaW"
                >
                  GRANT PROPOSAL
                </div>
              </div>
            </div>
            <div
              class="Card__CardTitle-h3ymxi-1 bSTxPd"
            >
              Tidbit Dynamic Oracles
            </div>
            <div
              class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
            >
              150000 Tokens Requested
            </div>
            <div
              class="Card__CardDescription-h3ymxi-3 iIGibR"
            >
              A proposal to make Ethereum safer by improving a reusable library for oracles.
            </div>
          </div>
          <div
            class="Card__Wrapper-h3ymxi-0 gjicgu"
          >
            <div
              class="flex"
            >
              <div
                class="Tag__Wrapper-l3n6sk-1 ceKKQx"
              >
                <div
                  class="Tag__StyledTag-l3n6sk-0 cZzgaW"
                >
                  GRANT PROPOSAL
                </div>
              </div>
            </div>
            <div
              class="Card__CardTitle-h3ymxi-1 bSTxPd"
            >
              Infrastructure Monitoring
            </div>
            <div
              class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
            >
              200000 Tokens Requested
            </div>
            <div
              class="Card__CardDescription-h3ymxi-3 iIGibR"
            >
              A proposal to make Ethereum safer by monitoring high value systems and assets on Ethereum.
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="flex justify-between"
      >
        <div
          class="flex"
        >
          <h1
            class="mr3 RouteTitle-ehvjdd-0 fHYele"
          >
            Proposals
          </h1>
          <a
            class="link flex items-center"
            href="/proposals/create"
          >
            <div
              class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
            >
              <button
                class="Button__StyledButton-sc-1vedn6s-0 izJlFT"
                type="default"
              >
                Add Proposal
              </button>
            </div>
          </a>
        </div>
        <div
          class="Deadline__Wrapper-sc-1a3aqlz-1 gtXKcM"
        >
          <div
            class="Tag__Wrapper-l3n6sk-1 ceKKQx"
          >
            <div
              class="Deadline__StyledDeadline-sc-1a3aqlz-0 jpSHwD Tag__StyledTag-l3n6sk-0 cZzgaW"
            >
              PROPOSAL DEADLINE Invalid date EST
            </div>
          </div>
        </div>
      </div>
      <div
        class="proposals__CardsWrapper-sc-125pbjx-0 cOJbkJ"
      >
        <div
          class="Card__Wrapper-h3ymxi-0 gjicgu"
        >
          <div
            class="flex"
          >
            <div
              class="Tag__Wrapper-l3n6sk-1 ceKKQx"
            >
              <div
                class="Tag__StyledTag-l3n6sk-0 cZzgaW"
              >
                GRANT PROPOSAL
              </div>
            </div>
          </div>
          <div
            class="Card__CardTitle-h3ymxi-1 bSTxPd"
          >
            K Semantics for Web Assembly
          </div>
          <div
            class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
          >
            200000 Tokens Requested
          </div>
          <div
            class="Card__CardDescription-h3ymxi-3 iIGibR"
          >
            A proposal to make Ethereum safer by enabling developers to formally verify smart contracts that run on Web Assembly.
          </div>
        </div>
        <div
          class="Card__Wrapper-h3ymxi-0 gjicgu"
        >
          <div
            class="flex"
          >
            <div
              class="Tag__Wrapper-l3n6sk-1 ceKKQx"
            >
              <div
                class="Tag__StyledTag-l3n6sk-0 cZzgaW"
              >
                GRANT PROPOSAL
              </div>
            </div>
          </div>
          <div
            class="Card__CardTitle-h3ymxi-1 bSTxPd"
          >
            Registry Builder Vote Weighting
          </div>
          <div
            class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
          >
            175000 Tokens Requested
          </div>
          <div
            class="Card__CardDescription-h3ymxi-3 iIGibR"
          >
            A proposal to make Ethereum safer by improving a reusable library for token-curated registries.
          </div>
        </div>
        <div
          class="Card__Wrapper-h3ymxi-0 gjicgu"
        >
          <div
            class="flex"
          >
            <div
              class="Tag__Wrapper-l3n6sk-1 ceKKQx"
            >
              <div
                class="Tag__StyledTag-l3n6sk-0 cZzgaW"
              >
                GRANT PROPOSAL
              </div>
            </div>
          </div>
          <div
            class="Card__CardTitle-h3ymxi-1 bSTxPd"
          >
            Tidbit Dynamic Oracles
          </div>
          <div
            class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
          >
            150000 Tokens Requested
          </div>
          <div
            class="Card__CardDescription-h3ymxi-3 iIGibR"
          >
            A proposal to make Ethereum safer by improving a reusable library for oracles.
          </div>
        </div>
        <div
          class="Card__Wrapper-h3ymxi-0 gjicgu"
        >
          <div
            class="flex"
          >
            <div
              class="Tag__Wrapper-l3n6sk-1 ceKKQx"
            >
              <div
                class="Tag__StyledTag-l3n6sk-0 cZzgaW"
              >
                GRANT PROPOSAL
              </div>
            </div>
          </div>
          <div
            class="Card__CardTitle-h3ymxi-1 bSTxPd"
          >
            Infrastructure Monitoring
          </div>
          <div
            class="Card__CardSubTitle-h3ymxi-2 kWUzrK"
          >
            200000 Tokens Requested
          </div>
          <div
            class="Card__CardDescription-h3ymxi-3 iIGibR"
          >
            A proposal to make Ethereum safer by monitoring high value systems and assets on Ethereum.
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
